Source/ThirdParty/libwebrtc/Source/webrtc/common_video/h265/h265_pps_parser.cc

```diff
@@ -53,20 +53,31 @@ bool H265PpsParser::ParsePpsIds(const uint8_t* data,
   RTC_DCHECK(sps_id);
   // First, parse out rbsp, which is basically the source buffer minus emulation
   // bytes (the last byte of a 0x00 0x00 0x03 sequence). RBSP is defined in
-  // section 7.3.1 of the H.265 standard.
-  std::vector<uint8_t> unpacked_buffer = H264::ParseRbsp(data, length);
-  BitstreamReader bit_buffer(unpacked_buffer);
-  return ParsePpsIdsInternal(&bit_buffer, pps_id, sps_id);
+  // section 7.3.1.1 of the H.265 standard.
+  std::vector<uint8_t> unpacked_buffer = H265::ParseRbsp(data, length);
+  BitstreamReader reader(unpacked_buffer);
+  *pps_id = reader.ReadExponentialGolomb();
+  *sps_id = reader.ReadExponentialGolomb();
+  return reader.Ok();
 }
 
 absl::optional<uint32_t> H265PpsParser::ParsePpsIdFromSliceSegmentLayerRbsp(
     const uint8_t* data,
     size_t length,
     uint8_t nalu_type) {
-  BitstreamReader slice_reader(rtc::MakeArrayView(data, length));
+  std::vector<uint8_t> unpacked_buffer = H265::ParseRbsp(data, length);
+  BitstreamReader slice_reader(unpacked_buffer);
+
+  // first_mb_in_slice: ue(v)
+  slice_reader.ReadExponentialGolomb();
+  // slice_type: ue(v)
+  slice_reader.ReadExponentialGolomb();
```

Comment:
This seems to be incorrect, slice header in h265 is defined as
![image](https://github.com/WebKit/WebKit/assets/1070835/e1143b5f-d6dd-4db1-9ee1-e363b4a90f82)

This seems to be following the one in h264 instead:
![image](https://github.com/WebKit/WebKit/assets/1070835/1127e94a-9580-4b6e-9f6f-23eb3cb2aabf)

Reply:
@murillo128 Thank you for the review. Will update the code based on the feedback shortly.

Reply:
updated.